query.ml

```diff
@@ -29,10 +29,15 @@ struct
   type query = comprehension list
 end
 
```

Comment:
It would be desirable to split query.ml into two modules, one that does shredding and the other one the standard/traditional one...